test/end2end_test.go

```diff
@@ -2121,6 +2121,74 @@ func testPreloaderClientSend(t *testing.T, e env) {
 	}
 }
 
+// preparedMsgSendServer is a TestServiceServer whose
+// StreamingOutputCall makes a SendMsg calls using PreparedMsg,
+// sending prepared messaged with payload "0".
+// TestPreloaderSenderSend verifies it is being sent correctly.
+//
+// All other TestServiceServer methods crash if called.
+type preparedMsgSendServer struct {
+	testpb.TestServiceServer
+}
+
+func (s preparedMsgSendServer) StreamingOutputCall(args *testpb.StreamingOutputCallRequest, stream testpb.TestService_StreamingOutputCallServer) error {
+	preparedMsg := &grpc.PreparedMsg{}
+	err := preparedMsg.Encode(stream, &testpb.StreamingOutputCallResponse{
+		Payload: &testpb.Payload{
+			Body: []byte{'0'},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	stream.SendMsg(preparedMsg)
+	return nil
+}
+
+func (s) TestPreloaderSenderSend(t *testing.T) {
+	for _, e := range listTestEnv() {
+		testPreloaderSenderSend(t, e)
+	}
```

Comment:
This isn't really the type of test we write anymore.  Please take a look at https://github.com/grpc/grpc-go/blob/4f3aa7cfa157c38bd5c2da7f4568614f815ab4ad/test/end2end_test.go#L5260 for an example.  This should hopefully make things a lot simpler (no need to make your own Server struct, e.g.)